front/lib/api/assistant/conversation.ts

```diff
@@ -1031,6 +1189,44 @@ export async function* editUserMessage(
     return;
   }
 
+  // Get new agent mentions, that are in 'mentions' array but not in message.mentions.
+  const additionalMentions = mentions.filter(
+    (mention) =>
+      isAgentMention(mention) &&
+      !message.mentions.some(
+        (m) =>
+          isAgentMention(m) && m.configurationId === mention.configurationId
+      )
+  ) as AgentMention[];
+
+  if (additionalMentions.length > 1) {
+    yield {
+      type: "user_message_error",
+      created: Date.now(),
+      error: {
+        code: "not_allowed",
+        message:
+          "Editing a message with more than 1 new agent mention is not yet supported",
```

Comment:
why?

Reply:
To go faster (re comment start simple on the issue)
hadn't assessed the effort to support multiple new mentions, just remembered the team chatting about the complexity of it (promise race, etc) and since we had said to start simple I went for that

No prob to work on changing it to multiple mentions if needed

Reply:
This is implemented right above in postUserMessage so we know how to do it.

The simple approach here is support many new mentions but only do so if it does not have children agent messages.

...
usermessage1
usermessage2

editing usermessage1 should trigger agents
editing usermesage2 should trigger agents

...
usermesage1
usermessage2
agentmessage(parent=2)
```

editing usermessage1 should trigger agents
editing usermesage2 should not trigger agents

Reply:
As per IRL, 
Backend: editUserMessage will run agents when 1. the user message is last 2. there were 0 mentions prior to the edit (i.e. all mentions are new mentions)
Frontend: we will have the agent suggestion **only for the last user message** for now

We can change the behaviour to "only for the user's last message" (so if other users posted message after, the suggestion is still there) quite simply if we accept that the agent messages will be appended at the end of the conversation